server/src/server.js

```diff
@@ -743,6 +743,30 @@ app.post("/api/set-title/:id/:domain", csrfProtection, function(req, res) {
   });
 });
 
+app.post("/api/save-edit", csrfProtection, function(req, res) {
+  let vars = req.body;
+  if (!req.deviceId) {
+    sendRavenMessage(req, "Attempt to set expiration without login");
+    simpleResponse(res, "Not logged in", 401);
+    return;
+  }
+  let id = vars.shotId;
+  let url = vars.url;
+  Shot.get(req.backend, id, req.deviceId, req.accountId).then((shot) => {
+    if (!shot) {
+      simpleResponse(res, "No such shot", 404);
```

Comment:
We should send a raven message here too, as it should be mostly impossible to get to this state (except maybe to delete a shot in a different tab while you are editing).